source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx

```diff
@@ -0,0 +1,100 @@
+import { FormikProps } from 'formik';
+import React from 'react';
+import { Redirect, Route, Switch, useRouteMatch } from 'react-router-dom';
+
+import { FileTypes } from '@/constants';
+import { InventoryTabNames, InventoryTabs } from '@/features/mapSideBar/property/InventoryTabs';
+import { FileTabType } from '@/features/mapSideBar/shared/detail/FileTabs';
+import { Api_AcquisitionFile } from '@/models/api/AcquisitionFile';
+
+import { UpdatePropertyDetailsContainer } from '../property/tabs/propertyDetails/update/UpdatePropertyDetailsContainer';
+import { TakesUpdateContainer } from '../property/tabs/takes/update/TakesUpdateContainer';
+import { TakesUpdateForm } from '../property/tabs/takes/update/TakesUpdateForm';
+import { PropertyFileContainer } from '../shared/detail/PropertyFileContainer';
+import { EditFormType } from './EditFormNames';
+
+export interface IFilePropertyRouterProps {
+  formikRef: React.Ref<FormikProps<any>>;
+  acquisitionFile?: Api_AcquisitionFile;
+  isEditing: boolean;
+  setIsEditing: (value: boolean) => void;
+  activeEditForm?: EditFormType;
+  selectedMenuIndex: number;
+  defaultFileTab: FileTabType;
+  defaultPropertyTab: InventoryTabNames;
+  onSuccess: () => void;
+}
+
+export const FilePropertyRouter: React.FC<IFilePropertyRouterProps> = props => {
+  const { path, url } = useRouteMatch();
+
+  if (props.acquisitionFile === undefined || props.acquisitionFile === null) {
+    return null;
+  }
+
+  const fileProperty = getAcquisitionFileProperty(props.acquisitionFile, props.selectedMenuIndex);
+
+  // render edit forms
+  if (props.isEditing) {
```

Comment:
same question here.

Reply:
I can throw an error like the old code used to do or redirect to file/property details if route doesn't match. Not sure what's better

Reply:
Fixed. We redirect to the default property tab (property details)